cmd/juju/commands/deploy.go

```diff
@@ -334,6 +335,10 @@ func (c *DeployCommand) deployCharmOrBundle(ctx *cmd.Context, client *api.Client
 	// Handle a charm.
 	curl, err := addCharmFromURL(client, charmOrBundleURL, repo, csClient)
 	if err != nil {
+		if err1, ok := err.(*termsRequiredError); ok {
```

Comment:
It's kind of ironic that we split them up only to join 'em together again...

You should use Cause here.

e.g.

      if err1, ok := errors.Cause(err).(*termsRequiredError); ok {